detekt/toolchain.bzl

```diff
@@ -0,0 +1,21 @@
+"""
+Detekt rules toolchain.
+"""
+
+def _impl(ctx):
+    toolchain = platform_common.ToolchainInfo(
+        jvm_flags = ctx.attr.jvm_flags,
+    )
+
+    return [toolchain]
+
+detekt_toolchain = rule(
+    implementation = _impl,
+    attrs = {
```

Comment:
hm, should we also move detekt jar into toolchain then?

Reply:
Yep, sounds good but doesn’t work in practice. As far as I understand the toolchain content cannot be resolved in a workspace-level macro, when we need to declare Maven dependencies. We need a context for that which is not available at this stage.